core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -1695,23 +1732,34 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord])
       this
     } else {
       val newElements = {
-        val nonWitUtxoVec =
-          this.getRecords(NonWitnessUTXOKeyId)
-        val witScriptVec =
-          this.getRecords(WitnessScriptKeyId)
-        if (nonWitUtxoVec.size == 1 && witScriptVec.size == 1) {
-          val nonWitUtxo = nonWitUtxoVec.head.transactionSpent
+        if (nonWitnessOrUnknownUTXOOpt.isDefined) {
+          val nonWitUtxo = nonWitnessOrUnknownUTXOOpt.get.transactionSpent
           if (txIn.previousOutput.vout.toInt < nonWitUtxo.outputs.size) {
             val out = nonWitUtxo.outputs(txIn.previousOutput.vout.toInt)
-            elements
-              .filterNot(_.key.head == NonWitnessUTXOKeyId.byte)
-              .filterNot(_.key.head == WitnessUTXOKeyId.byte) :+ WitnessUTXO(
-              out)
+            out.scriptPubKey match {
+              case _: WitnessScriptPubKey =>
+                sortedElements
+                  .filterNot(_.key.head == NonWitnessUTXOKeyId.byte)
+                  .filterNot(_.key.head == WitnessUTXOKeyId.byte) :+ WitnessUTXO(
```

Comment:
We should probably require that you can't have both `WitnessUTXO` and `NonWitnessUTXO` populated, then you can get rid of the `WitnessUTXO` filter here